test/tools/libtesteth/ImportTest.cpp

```diff
@@ -289,11 +296,18 @@ json_spirit::mObject& ImportTest::makeAllFieldsHex(json_spirit::mObject& _o, boo
 			str = value.get_str();
 		else if (value.type() == json_spirit::array_type)
 		{
+			//convert transaction value, gaslimit arrays
 			json_spirit::mArray arr;
-			for (auto const& j: value.get_array())
+			for (auto& j: value.get_array())
 			{
-				str = j.get_str();
-				arr.push_back((str.substr(0, 2) == "0x") ? str : toCompactHexPrefixed(toInt(str), 1));
+				if (j.type() == json_spirit::str_type)
+				{
+					str = j.get_str();
+					str = (str.substr(0, 2) == "0x") ? str : toCompactHexPrefixed(toInt(str), 1);
+					arr.push_back(str);
+				}
+				else
+					arr.push_back(j);
```

Comment:
Now I'm confused.  Naturally I would make `postParseJson()` call `makeAllFieldsHex()` and `makeAllFieldsHex()` call `postParseJson()`.  But it seems like something else is happening.